File: csharp/DVSClient/DVSClient/address/datasets/GetDatasetsResult.cs
using DVSClient.Common;
using DVSClient.Server.Address.Datasets;

namespace DVSClient.Address.Datasets
{
    public class GetDatasetsResult
    {
        public ResponseError? Error { get; }
        public IEnumerable<AddressDataset>? Result { get; }

        public GetDatasetsResult(RestApiGetDatasetsResponse response)
        {
            Error = response.Error != null ? new ResponseError(response.Error) : null;
            Result = response.Result != null ? response.Result.Select(r => new AddressDataset(r)).ToList() : null;
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/datasets/GetDatasetsResult.java
package com.experian.dvs.client.address.datasets;

import com.experian.dvs.client.server.address.dataset.RestApiGetDatasetsResponse;

import java.util.List;

public class GetDatasetsResult {

    private final List<AddressDataset> result;
    private final String referenceId;

    public GetDatasetsResult(final RestApiGetDatasetsResponse response) {
        final var res = response.getResult();
        if (res != null) {
            this.result = response.getResult().stream().map(AddressDataset::new).toList();
        } else {
            this.result = null;
        }

        this.referenceId = response.getReferenceId();
    }

    public List<AddressDataset> getResult() {
        return result;
    }

    public String getReferenceId() {
        return referenceId;
    }
}
